src/includes_normalize-win32.cc

```diff
@@ -28,16 +28,26 @@ namespace {
 
 bool InternalGetFullPathName(const StringPiece& file_name, char* buffer,
                              size_t buffer_length, string *err) {
-  DWORD result_size = GetFullPathNameA(file_name.AsString().c_str(),
-                                       buffer_length, buffer, NULL);
+  wchar_t w_buffer[_MAX_PATH];
+  wstring w_file_name = Utf8ToWide(file_name.AsString());
```

Comment:
Why the call to AsString()? Utf8ToWide doesn't actually care that you give it a string (see my comment about using StringPiece). This seems like an unnecessary allocation and copy.

Reply:
Also

std::wstring w_file_name.

Newly added code should not discard the std:: namespace.